cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -670,7 +670,6 @@ void AddDecimalBinaryKernels(const std::string& name, ScalarFunction* func) {
     out_type = OutputType(ResolveDecimalMultiplicationOutput);
   } else if (op == "divide") {
     out_type = OutputType(ResolveDecimalDivisionOutput);
-    constraint = BinaryDecimalScale1GeScale2();
```

Comment:
We don't really need this constraint to suppress the exact matching as this is now done via overridden `DispatchExact`.

Reply:
Why not a `DecimalsHaveSameScaleAndPrecision`? (or full type equality, which is exactly equivalent here)

Reply:
Well, `{decimal(5, 1), decimal(5, 1)}` looks like an exact match in this example. The result type is unrelated to this.

Reply:
Hmm, thanks. Perhaps the PR description can be clearer about this?

Reply:
And this is how we obey the resulting type rule we claim - promoting the dividend.

That said, there is an alternative though - as you implied in your previous comment
> looks like an exact match in this example. The result type is unrelated to this.

This is also explained in my PR description approach 2. I didn't take that approach because that would require the promotion to happen during the underlying division for each individual value in the array. Can be cumbersome in terms of both coding and performance.

Reply:
Ok, thanks for the explanation!